test/unit_tests/braket/experimental/autoqasm/test_parameters.py

```diff
@@ -484,5 +486,264 @@ def test_binding_variable_fails():
     def parametric():
         alpha = aq.FloatVar(1.2)  # noqa: F841
 
-    with pytest.raises(ValueError, match="No parameter in the program named: beta"):
+    with pytest.raises(
+        aq.errors.ParameterNotFoundError, match="No parameter in the program named: beta"
+    ):
         parametric().make_bound_program({"beta": 0.5}, strict=True)
+
+
+def test_compound_condition():
+    """Test parameters used in greater than conditional statements."""
+
+    @aq.main
+    def parametric(val: float):
+        threshold = 0.9
+        if val > threshold or val >= 1.2:
+            x(0)
+        measure(0)
+
+    expected = """OPENQASM 3.0;
+input float[64] val;
+qubit[1] __qubits__;
+bool __bool_0__;
+__bool_0__ = val > 0.9;
+bool __bool_1__;
+__bool_1__ = val >= 1.2;
+bool __bool_2__;
+__bool_2__ = __bool_0__ || __bool_1__;
+if (__bool_2__) {
+    x __qubits__[0];
+}
+bit __bit_3__;
+__bit_3__ = measure __qubits__[0];"""
+    assert parametric(FreeParameter("val")).to_ir() == expected
+
+
+def test_lt_condition():
+    """Test parameters used in less than conditional statements."""
+
+    @aq.main
+    def parametric(val: float):
+        if val < 0.9:
+            x(0)
+        if val <= 0.9:
+            h(0)
+        measure(0)
+
+    expected = """OPENQASM 3.0;
+input float[64] val;
+qubit[1] __qubits__;
+bool __bool_0__;
+__bool_0__ = val < 0.9;
+if (__bool_0__) {
+    x __qubits__[0];
+}
+bool __bool_1__;
+__bool_1__ = val <= 0.9;
+if (__bool_1__) {
+    h __qubits__[0];
+}
+bit __bit_2__;
+__bit_2__ = measure __qubits__[0];"""
+    assert parametric(FreeParameter("val")).to_ir() == expected
+
+
+def test_parameter_in_predicate_in_subroutine():
+    """Test parameters used in conditional statements."""
+
+    @aq.subroutine
+    def sub(val: float):
+        threshold = 0.9
+        if val > threshold:
+            x(0)
+
+    @aq.main
+    def parametric(val: float):
+        sub(val)
+        measure(0)
+
+    expected = """OPENQASM 3.0;
+def sub(float[64] val) {
+    bool __bool_0__;
+    __bool_0__ = val > 0.9;
+    if (__bool_0__) {
+        x __qubits__[0];
+    }
+}
+input float[64] val;
+qubit[1] __qubits__;
+sub(val);
+bit __bit_1__;
+__bit_1__ = measure __qubits__[0];"""
+    assert parametric(FreeParameter("val")).to_ir() == expected
+
+
+def test_eq_condition():
+    """Test parameters used in conditional equals statements."""
+
+    @aq.main
+    def parametric(basis: int):
+        if basis == 1:
+            h(0)
+        elif basis == 2:
+            x(0)
+        else:
+            pass
+        measure(0)
+
+    expected = """OPENQASM 3.0;
+input float[64] basis;
+qubit[1] __qubits__;
+bool __bool_0__;
+__bool_0__ = basis == 1;
+if (__bool_0__) {
+    h __qubits__[0];
+} else {
+    bool __bool_1__;
```

Comment:
Does this work with simulator? Per https://github.com/amazon-braket/amazon-braket-sdk-python/pull/650 we biased toward not having variable declare in if-else scope. @rmshaffer do you remember why we didn't want declaration in the if-else scope? Was it solely because of simulator support?

Reply:
For user-defined variables in Python, they can be separately assigned in `if` and `else` blocks like:
```
if cond:
    a = aq.IntVar(1)
else:
    a = aq.IntVar(2)
return a
```
In the OpenQASM the `a` variable needs to be declared prior to the `if` block in order for this to work.

In this case since it is an auto-generated variable that only exists in the OpenQASM, it will not be used in multiple Python blocks, so I think there should be no issue declaring it here.

Reply:
I see, it was the declaration inside if branch being in different scope of else branch. Nested scope should work.

Reply:
I'll add some tests with the simulator

Reply:
Added new tests!